packages/ra-core/src/controller/details/CreateContext.tsx

```diff
@@ -0,0 +1,49 @@
+import { createContext, useContext } from 'react';
+import { Record } from '../../types';
+import { CreateControllerProps } from './useCreateController';
+
+/**
+ * Context to store the result of the useCreateController() hook.
+ *
+ * Use the useCreateContext() hook to read the context. That's what the Create components do in react-admin.
+ *
+ * @example
+ *
+ * import { useCreateController, CreateContext } from 'ra-core';
+ *
+ * const Create = props => {
+ *     const controllerProps = useCreateController(props);
+ *     return (
+ *         <CreateContext.Provider value={controllerProps}>
+ *             ...
+ *         </CreateContext.Provider>
+ *     );
+ * };
+ */
+export const CreateContext = createContext<CreateControllerProps>({
+    basePath: null,
+    record: null,
+    defaultTitle: null,
+    loaded: null,
+    loading: null,
+    redirect: null,
+    setOnFailure: null,
+    setOnSuccess: null,
+    setTransform: null,
+    resource: null,
+    save: null,
+    saving: null,
+    successMessage: null,
+    version: null,
+});
+
+CreateContext.displayName = 'CreateContext';
+
+export const useCreateContext = <
+    RecordType extends Record = Record
+>(): CreateControllerProps<RecordType> => {
```

Comment:
See what I did in useListContext: if the context is empty, we fallback to props. This allows for a smooth transition.

```
const useListContext = <RecordType extends Record = Record>(
    props?: any
): ListControllerProps<RecordType> => {
    const context = useContext(ListContext);
    if (!context.resource) {
        /**
         * The element isn't inside a <ListContext.Provider>
         *
         * This may only happen when using Datagrid / SimpleList / SingleFieldList components
         * outside of a List / ReferenceManyField / ReferenceArrayField -
         * which isn't documented but tolerated.
         * To avoid breakage in that case, fallback to props
         *
         * @deprecated - to be removed in 4.0
         */
        if (process.env.NODE_ENV !== 'production') {
            console.log(
                "List components must be used inside a <ListContext.Provider>. Relying on props rather than context to get List data and callbacks is deprecated and won't be supported in the next major version of react-admin."
            );
        }
        return props;
    }
    // @ts-ignore
    return context;
};
```